src/workerd/api/node/i18n.c++

```diff
@@ -0,0 +1,158 @@
+#include "i18n.h"
+
+#include <kj/common.h>
+#include <kj/debug.h>
+#include <kj/one-of.h>
+
+#include <workerd/jsg/exception.h>
+
+#include <string>
+
+namespace workerd::api::node {
+
+namespace i18n {
+
+namespace {
+
+const char* getEncodingName(Encoding input) {
+  switch (input) {
+  case Encoding::ASCII:
+    return "us-ascii";
+  case Encoding::LATIN1:
+    return "iso8859-1";
+  case Encoding::UCS2:
+    return "utf16le";
+  case Encoding::UTF8:
+    return "utf-8";
+  default:
+    KJ_UNREACHABLE;
+  }
+}
+
+typedef TranscodeResult (*TranscodeImpl)(kj::ArrayPtr<kj::byte> source, Encoding fromEncoding,
+                                         Encoding toEncoding);
+
+TranscodeResult TranscodeDefault(kj::ArrayPtr<kj::byte> source, Encoding fromEncoding,
+                                 Encoding toEncoding) {
+  Converter to(toEncoding);
+  std::string substitude(to.minSize(), '?');
+  to.setSubstitudeChars(substitude);
+
+  Converter from(fromEncoding);
+
+  auto limit = source.size() + to.maxSize();
+  KJ_STACK_ARRAY(kj::byte, out, limit, 0, limit);
+  char* target = out.asChars().begin();
+  const char* source_ = source.asChars().begin();
+  UErrorCode status{};
+  ucnv_convertEx(to.conv(), from.conv(), &target, target + limit, &source_, source_ + source.size(),
+                 nullptr, nullptr, nullptr, nullptr, true, true, &status);
+
+  if (U_SUCCESS(status)) {
+    return out.slice(0, target - out.asChars().begin()).attach();
```

Comment:
This is going to lead to a UAF, I believe. The `out` here is stack allocated and you're returning an `ArrayPtr` from it. I don't believe the `attach()` call here causes it to be copied at all.

I think what you want here is `auto out = kj::heapArray<kj::byte>(limit)` instead of `KJ_STACK_ARRAY` and then `return out.slice(0, target - out.begin()).attach(kj::mv(out))`